File: Second_Semester/Discrete-Structure/Recursive-Problems/factorial.c
//Wap to implement recursive algorithm to find factorial of non-ngative integer
#include <stdio.h>
int fact(int);
void main(){
	int n,f;
	printf("Enter a number to find factorial: ");
	scanf("%d",&n);
	if(n<0){
		printf("Negative number do not have factorial.");
	}
	else{
		f = fact(n);
		printf("Factorial of %d is %d.",n,f);
	}
}
int fact(int n){
	int i;
	if(n==0){
		return 1;
	}
	else{
		return n*fact(n-1);
	}
}

File: Third_Semester/Data-Structure-And-Algorithms-with-JAVA/recursion/Factorial.java

package recursion;
import java.util.Scanner;

public class Factorial {
    static Scanner sc = new Scanner (System.in);
    
    int factorial(int n){
        if(n==0){ //base case
            return 1;
        }
        return n*factorial(n-1);
    }

    public static void main(String[] args) {
        Factorial fact = new Factorial();
        System.out.println("Enter a number to find factorial");
        int n = sc.nextInt();
        System.out.println("Factorial of "+n+" is "+fact.factorial(n));
    }
}
